aten/src/ATen/native/TensorIterator.cpp

```diff
@@ -95,15 +109,11 @@ void TensorIterator::compute_common_type() {
   for (auto& op : operands_) {
     if (!op.type) {
       op.type = &type;
-      if (op.tensor->defined() && type != op.tensor->type()) {
-        if (op.tensor->dim() == 0) {
-          if (type.backend() != at::Backend::CUDA) {
-            cast_tensors_.emplace_back(op.tensor->toType(type));
-            op.tensor = &(cast_tensors_.back());
-          }
-        } else {
-          op.needs_cast = true;
-        }
+      op.needs_cast = needs_cast(*op.tensor, type);
+      if (op.needs_cast && op.tensor->dim() == 0) {
```

Comment:
What's the reason it only casts for 0 dim tensors and not for any number of dimensions?

Reply:
Eventually, we'll allow casts for any number of dimensions probably using the logic described here:
https://github.com/pytorch/pytorch/issues/9515

For now, we only support casts for zero-dim tensors because it's the minimum necessary to preserve backwards compatibility.